### DISCUSSION TITLE
Log for nginx limit req

### DISCUSSION THREAD

**Discussion by NoipName** [fail2ban/fail2ban#3679] [Q&A] [answered]
Hello everyone,
I installed, fail2ban on my server but i have somequestion for the configuration of fail2ban
My page web generated ~200 request/s and my configuration fail2ban is zone one = 30Mo and 500r/s

Burst = 50

With this configuration i have pretty much "delayed" and "limiting" in my log
![image](https://github.com/fail2ban/fail2ban/assets/145665195/e435bebf-1ede-42b8-a156-6ce1479a38f7)

Someone can explain me this ?

Thanks you guys

**Comment by sebres** [comment, marked as answer]
Firstly, this has nothing to do with fail2ban.

Related to nginx web-pages:
*If the number of requests exceeds the specified rate, NGINX will delay processing of such requests until the “bucket” is full. For requests that arrive at the full bucket, NGINX will respond with the 503 Service Unavailable error (if not redefined with limit_req_status)*

See [Limiting Access to Proxied HTTP Resources - NGINX Docs](https://docs.nginx.com/nginx/admin-guide/security-controls/controlling-access-proxied-http/#:~:text=If%20the%20number%20of%20requests,if%20not%20redefined%20with%20limit_req_status),
[NGINX Rate Limiting](https://www.nginx.com/blog/rate-limiting-nginx/)
and [Module ngx_http_limit_req_module](http://nginx.org/en/docs/http/ngx_http_limit_req_module.html)

As for `nginx-limit-req.conf` filter: it would only find by limit terminated requests only (requests their number exceeds the maximum burst size):
https://github.com/fail2ban/fail2ban/blob/dce2c608c19e1907bdeefc81befdd52181930238/config/filter.d/nginx-limit-req.conf#L49

The failregex can be surely extended to match delayed requests too, but it'd be a bit weird constellation - why one would wish to ban them on fail2ban side, but delay on nginx side? This is pretty inconsistent.

**Reply by NoipName** [reply]
Hello, thanks you for explication
I edited my configuration and add the argument "nodelay" in my host file.
Now i dont have "delaying" notice anymore

Thanks you :D

**Reply by sebres** [reply]
> Now i dont have "delaying" notice anymore

Well, just for the record (I am unsure you understand that correctly) - I did not claimed that your nginx-config was not alright.
I just said, it is not necessary to consider delay messages as a failure (a reason for ban).
Even without `nodelay` it works OK, in the way that an IP will be simply delayed until the “bucket” is full, and then after it'd log `limiting requests` it would considered as a failure by fail2ban.

If you just want not that delay messages get logged, just decrease it via the `limit_req_log_level` directive.
As [doku for limit_req_log_level](https://nginx.org/en/docs/http/ngx_http_limit_req_module.html#limit_req_log_level) says:
*Logging level for delays is one point less than for refusals; for example, if `limit_req_log_level notice` is specified, delays are logged with the `info` level.*

Thus take a look at your [error_log](https://nginx.org/en/docs/ngx_core_module.html#error_log), to configure `limit_req_log_level` properly, for instance if if `limit_req_log_level error` is specified (it is also default), and `error_log ... error` is specified too (also default level), delays should be then logged with the `warn` level (and it would be not logged, since filtered).
In any case to avoid delayed messages in log, the value of `limit_req_log_level` should match your `error_log` level.
The levels of error-log are: `debug`, `info`, `notice`, `warn`, `error`, `crit`, `alert`, `emerg` (from smaller to larger point).

Anyway forcing of delay by limits is alright, but it is your decision how you'd configure it.
